types/datum.go

```diff
@@ -914,33 +914,17 @@ func (d *Datum) convertToMysqlTimestamp(sc *stmtctx.StatementContext, target *Fi
 	if target.Decimal != UnspecifiedLength {
 		fsp = target.Decimal
 	}
-	loc := sc.TimeZone
 	switch d.k {
 	case KindMysqlTime:
 		t = d.GetMysqlTime()
-		if t.Type == mysql.TypeTimestamp {
```

Comment:
Why?

Reply:
If the input datum is not already in session's TimeZone, it maybe wrong here.
But I'm not sure when it will happen that datum's TimeZone != session's TimeZone.